plutus-scb/src/Cardano/Protocol/Socket/Type.hs

```diff
@@ -0,0 +1,109 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DerivingStrategies    #-}
+{-# LANGUAGE DerivingVia           #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE TypeFamilies          #-}
+{-# OPTIONS -Wno-orphans #-} -- TODO: remove this
+module Cardano.Protocol.Socket.Type where
+
+import           Codec.Serialise.Class                              (Serialise)
+import           Data.Aeson                                         (FromJSON, ToJSON)
+import qualified Data.ByteString.Lazy                               as BSL
+import           Data.Text.Prettyprint.Doc                          (Pretty)
+
+import           Cardano.Prelude                                    (NoUnexpectedThunks)
+import           GHC.Generics
+
+import           Codec.Serialise                                    (DeserialiseFailure)
+import qualified Codec.Serialise                                    as CBOR
+import           Network.TypedProtocol.Codec
+import           Ouroboros.Network.Block                            (HeaderHash, StandardHash)
+import           Ouroboros.Network.Mux
+import qualified Ouroboros.Network.Protocol.ChainSync.Codec         as ChainSync
+import qualified Ouroboros.Network.Protocol.ChainSync.Type          as ChainSync
+import qualified Ouroboros.Network.Protocol.LocalTxSubmission.Codec as TxSubmission
+import qualified Ouroboros.Network.Protocol.LocalTxSubmission.Type  as TxSubmission
+
+import           Ledger                                             (Block, Tx (..), TxId (..), txId)
+import qualified Ledger.Index                                       as Index
+import           LedgerBytes                                        (LedgerBytes (..))
+import           Wallet.Emulator.Chain
+
+-- | Instances used for CBOR serialisation.
+deriving newtype instance CBOR.Serialise Index.UtxoIndex
+deriving instance Generic ChainState
+deriving instance Serialise ChainState
```

Comment:
I think these can go with the type definitions, pretty sure `serialise` is already a dependency everywhere.